lib/ae_mdw/oracles.ex

```diff
@@ -16,38 +16,60 @@ defmodule AeMdw.Oracles do
   # This needs to be an actual type like AeMdw.Db.Oracle.t()
   @type oracle :: term()
   @typep limit :: Mnesia.limit()
+  @typep scope :: {:gen, Range.t()} | nil
 
   @table_active AeMdw.Db.Model.ActiveOracle
   @table_active_expiration Model.ActiveOracleExpiration
   @table_inactive AeMdw.Db.Model.InactiveOracle
   @table_inactive_expiration Model.InactiveOracleExpiration
 
-  @spec fetch_oracles(Mnesia.direction(), cursor() | nil, limit(), boolean()) ::
+  @spec fetch_oracles(Mnesia.direction(), scope(), cursor() | nil, limit(), boolean()) ::
           {[oracle()], cursor() | nil}
-  def fetch_oracles(direction, cursor, limit, expand?) do
+  def fetch_oracles(direction, scope, cursor, limit, expand?) do
     cursor = deserialize_cursor(cursor)
 
-    {{start_table, start_keys}, {end_table, end_keys}, next_key} =
-      Collection.concat(
-        @table_inactive_expiration,
-        @table_active_expiration,
-        direction,
-        cursor,
-        limit
-      )
+    gen_range =
+      case scope do
+        nil ->
+          nil
+
+        {:gen, %Range{first: first_gen, last: last_gen}} ->
+          if direction == :forward do
+            {{first_gen, <<>>}, {last_gen, <<>>}}
+          else
+            {{first_gen, nil}, {last_gen, nil}}
+          end
+      end
```

Comment:
The naming used in `AeMdw.Oracles` and `AeMdw.Names` looks conflicting with the one in AeMdw.Collection regarding to the `scope()` in a way they appear swapped. Let me try to explain ...

One might expected `gen_range` variable to be a `{:gen, %Range{first: first_gen, last: last_gen}}` while scope to be the `{{first_gen, <<>>}, {last_gen, <<>>}}` or `{{first_gen, nil}, {last_gen, nil}}` because `scope()` type of the `AeMdw.Collection.stream` `scope` argument is a `{key, key}`.

Another factor contributes to make this a good opportunity for a better readability is in regards to definition of `scope()` type of `AeMdw.Collection`. Looking to the :oracle and :tx models, neither of their keys are like `{first_gen, <<>>} or {first_gen, nil}`.

Reply:
Done